lib/core/test/unit/Cardano/Wallet/DB/StateMachine.hs

```diff
@@ -526,6 +534,18 @@ generator (Model _ wids) = Just $ frequency $ fmap (fmap At) <$> concat
     genPrivKey :: Gen MPrivKey
     genPrivKey = elements ["pk1", "pk2", "pk3"]
 
+    genSortDir :: Gen SortDirection
+    genSortDir = QC.oneof [return Ascending, return Descending]
+
+    genFilter :: Gen (Filter SlotId)
+    genFilter = Filter <$> genSId <*> genSId
```

Comment:
So this potentially generates invalid ranges :thinking: ?

Reply:
Yes. This PR allows unsatisfiable filters like`x > 10 && x < 2`. 

I originally thought that would be fine… but maybe we should do something… 
1. `unsafeMkFilter :: Maybe a -> Maybe a -> Filter a -- throws` 
2. `mkFilter :: Maybe a -> Maybe a -> Maybe (Filter a)` 
3. `mkFilter :: Maybe a -> Maybe a -> Filter a -- may swap the arguments such that low <= high is always true`

But I don't really like any of them. Maybe we are fine with doing nothing.

Also I'm not sure I'm totally thrilled about simply calling this type `Filter`.

Reply:
> Also I'm not sure I'm totally thrilled about simply calling this type `Filter`.

Maybe just call it `Range`, given the comment:

```hs
-- Represents the range [start, end]
```